MetaMorpheus/EngineLayer/Indexing/IndexingEngine.cs

```diff
@@ -258,9 +260,19 @@ private void AddInteriorTerminalModsToPrecursorIndex(List<int>[] precursorIndex,
             foreach (KeyValuePair<int, List<Modification>> relevantDatabaseMod in databaseAnnotatedMods)
             {
                 int fragmentNumber = relevantDatabaseMod.Key;
-                Product fragmentAtIndex = fragmentMasses.Where(x => x.FragmentNumber == fragmentNumber).FirstOrDefault();
```

Comment:
In the previous version, default Proudct would produce an a1 product with a mass of 0.0